src/components/AttachmentModal.js

```diff
@@ -310,7 +325,7 @@ function AttachmentModal(props) {
                 <HeaderWithBackButton
                     title={props.headerTitle || props.translate('common.attachment')}
                     shouldShowBorderBottom
-                    shouldShowDownloadButton={props.allowDownload}
+                    shouldShowDownloadButton={props.allowDownload && shouldShowDownloadButton}
```

Comment:
I am wondering if this can be just replaced with `state`, and `props.allowDownload` can be used in the useCallback?

Reply:
I think using `props.allowDownload` in `useCallback` dependencies will trigger change definition for method `setDownloadButtonVisibility` not  `shouldShowDownloadButton` value.

but we can use it in useEffect.
```javascript
const [shouldShowDownloadButton, setShouldShowDownloadButton] = React.useState(props.allowDownload);

useEffect(() => {
  if (shouldShowDownloadButton === props.allowDownload) {
      return;
  }
  setShouldShowDownloadButton(props.allowDownload);
}, [props.allowDownload]);
```

what do you think? @mananjadhav

Reply:
I think we can ignore this change.